### ISSUE TITLE
Bug: Disabled DHCP interfaces are enabled on dhcpd.conf (patch provided)

### ISSUE THREAD

**Issue by patphobos** [zentyal/zentyal#1903]
Hi,

this is a patch to only generate configuration of enabled dhcp interfaces.
because pre-configured interfaces with ranges configured are inserted in dhcpd.conf even if the interface is marked has "disabled".

patch for : /usr/share/perl5/EBox/DHCP.pm
```perl
--- EBoxOrig/DHCP.pm    2018-09-05 11:12:04.000000000 +0200
+++ EBoxPatched/DHCP.pm 2019-10-30 17:43:10.768880250 +0100
@@ -737,6 +737,16 @@
     return $model->addresses($iface, $readOnly);
 }

+sub dhcpEnabled
+{
+    my ($self, $iface) = @_;
+
+    my $ifaceModel = $self->model('Interfaces');
+    my $ifaceRow = $ifaceModel->findRow(iface => $iface);
+    return $ifaceRow->valueByName('enabled');
+}
+
+
 # Group: Static or class methods

 # Method: ConfDir
@@ -1187,6 +1197,8 @@
             my $address = $net->ifaceAddress($iface);
             my $netmask = $net->ifaceNetmask($iface);
             my $network = ip_network($address, $netmask);
+
+            $iflist{$iface}->{'enabled'} = $self->dhcpEnabled($iface);

             $iflist{$iface}->{'net'} = $network;
             $iflist{$iface}->{'address'} = $address;
```

patch for dhcpd.conf.mas : 
```
--- orig/dhcpd.conf.mas        2018-09-05 11:12:04.000000000 +0200
+++ patched/dhcpd.conf.mas        2019-10-30 17:41:09.252997815 +0100
@@ -71,7 +71,7 @@
 <%perl>
             # Write configuration if there are any range or fixed addresses
             my $ifaceToWrite = 0;
-            if (exists $ifaces{$iface}) {
+            if (exists $ifaces{$iface} && $ifaces{$iface}->{enabled}) {
                 if (@{$ifaces{$iface}->{ranges}} > 0) {
                     $ifaceToWrite = 1;
                 } elsif  (values %{$ifaces{$iface}->{fixed}} > 0 ) {
```

**Comment by Neustradamus** [comment]
@brunolorente: Have you seen this ticket and the PR?

**Comment by brunolorente** [comment]
Hi @patphobos ,

Many thanks for your contribution, your change was added to Zentyal 6.1 and Zentyal 6.2. Fixed in 064e2c9825561015d8202888553026b28959e523
